src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+        }
+        assert car != null;
+        car.setDrivers(getCarDrivers(id));
+        return Optional.of(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name,"
+                + " manufacturers.country AS manufacturer_country, model "
+                + "FROM cars c JOIN manufacturers  ON c.manufacturer_id = manufacturers.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+        cars.forEach(car -> car.setDrivers(getCarDrivers(car.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement
+                        = connection.prepareStatement(query)) {
+            updateCarStatement.setLong(1, car.getManufacturer().getId());
+            updateCarStatement.setString(2, car.getModel());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        removeDriversFromCar(car);
+        addCarDrivers(car.getDrivers(), car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, manufacturer_id, model "
+                + "FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id"
+                + " WHERE cd.driver_id = ? AND c.is_deleted = FALSE;";
```

Comment:
You also should check if the driver is not deleted

Reply:
there is check ...AND c.is_deleted = FALSE;

Reply:
yes, you check that car is not deleted but the driver  also can be deleted 
so you need something  like this AND d.is_deleted = FALSE (you will also need to join a driver table)